starknet_py/tests/e2e/client/conftest.py

```diff
@@ -1,57 +1,54 @@
+# pylint: disable=redefined-outer-name
+
 import os
-import re
 import subprocess
 from pathlib import Path
-from ast import literal_eval
 from typing import Tuple
 
 import pytest
+import pytest_asyncio
 from starkware.starknet.public.abi import get_selector_from_name
 
+from starknet_py.net import AccountClient
 from starknet_py.net.client import Client
 from starknet_py.net.full_node_client import FullNodeClient
 from starknet_py.net.gateway_client import GatewayClient
 from starknet_py.net.models import StarknetChainId
+from starknet_py.tests.e2e.client.prepare_net_for_gateway_test import (
+    prepare_net_for_tests,
+)
 from starknet_py.tests.e2e.conftest import directory_with_contracts
 
 directory = os.path.dirname(__file__)
 
 
-def prepare_devnet(net: str) -> dict:
-    script_path = Path(directory) / "prepare_devnet_for_gateway_test.sh"
-    contract_compiled = directory_with_contracts / "balance_compiled.json"
-    contract_abi = directory_with_contracts / "balance_abi.json"
-
-    res = subprocess.run(
-        [script_path, net, contract_compiled, contract_abi],
-        check=False,
-        capture_output=True,
-        text=True,
+async def prepare_network(gateway_account_client: AccountClient) -> dict:
+    contract_compiled = Path(
+        directory_with_contracts / "balance_compiled.json"
+    ).read_text("utf-8")
+
+    (
+        contract_address,
+        deploy_transaction_hash,
+        block_with_deploy_number,
+        block_with_deploy_hash,
+        invoke_transaction_hash,
+        block_with_invoke_number,
+        declare_transaction_hash,
+        block_with_declare_number,
+    ) = await prepare_net_for_tests(
+        gateway_account_client, compiled_contract=contract_compiled
     )
 
-    genesis_block = res.stdout.splitlines()[-2]
-    genesis_block = literal_eval(genesis_block)
-    block = res.stdout.splitlines()[-1]
-    block = literal_eval(block)
-    contract_address = res.stdout.splitlines()[2].split(sep=" ")[-1]
-    deploy_transaction_hash = res.stdout.splitlines()[3].split(sep=" ")[-1]
-    invoke_transaction_hash = res.stdout.splitlines()[6].split(sep=" ")[-1]
-    declare_transaction_hash = res.stdout.splitlines()[9].split(sep=" ")[-1]
-
-    assert genesis_block != ""
-    assert block != ""
-    assert re.search("^0x0[a-fA-F0-9]{1,63}$", contract_address) is not None
-    assert re.search("^0x[a-fA-F0-9]{1,63}$", deploy_transaction_hash) is not None
-    assert re.search("^0x[a-fA-F0-9]{1,63}$", invoke_transaction_hash) is not None
-    assert re.search("^0x[a-fA-F0-9]{1,63}$", declare_transaction_hash) is not None
-
     prepared_data = {
-        "genesis_block": genesis_block,
-        "block": block,
-        "contract_address": int(contract_address, 16),
-        "deploy_transaction_hash": int(deploy_transaction_hash, 16),
+        "contract_address": contract_address,
```

Comment:
What's the contract address?

Reply:
It's an address of deployed balance contract

Reply:
Do we use any specific functions from it? If so we should name it "balance_contract_address".

Reply:
No, we don't. I've checked that